pygpcca/utils/_checks.py

```diff
@@ -122,3 +123,28 @@ def ensure_ndarray_or_sparse(
     assert_array(A, shape=shape, uniform=uniform, ndim=ndim, size=size, dtype=dtype, kind=kind)
 
     return A
+
+
+def petsc_real_scalar_type() -> bool:
+    """Check PETSc was compiled using `–with-scalar-type=real`."""
+    try:
+        from petsc4py import PETSc
+
+        if np.isrealobj(PETSc.ScalarType()):
+            return True
+        raise TypeError("Complex scalar type is not supported.")
+    except ImportError:
+        return False
```

Comment:
@michalk8, guess I would return True here if ImportError, since this is handled elsewhere and should not trigger scalar type error.

Reply:
Ok, I'd then refactor as `assert_petsc_scalar_type()` which raises the `TypeError` if not scalar and if it cannot import, doesn't raise anything. WDYT?